### ISSUE TITLE
InvalidArgumentException: Cache frontend 'default' is not recognized

### ISSUE THREAD

**Issue by ivoinov** [magento/magento2#13535]
### Preconditions

1. Magento 2.1.9
2. PHP 7.0.27. Mysql 5.6.

### Steps to reproduce
1. Switched default and page cache from file to Redis cache.
`    'cache'           => array(
        'frontend' => array(
            'default'    => array(
                'backend'         => 'Cm_Cache_Backend_Redis',
                'backend_options' => array(
                    'server'   => '172.19.0.3',
                    'database' => '0',
                    'port'     => '6379',
                ),
            ),
            'page_cache' => array(
                'backend'         => 'Cm_Cache_Backend_Redis',
                'backend_options' => array(
                    'server'        => '172.19.0.3',
                    'port'          => '6379',
                    'database'      => '1',
                    'compress_data' => '0',
                ),
            ),
        ),
    ),`

### Expected result
No error occurs

### Actual result
Class Magento\Support\Model\Logger\Handler\Report does not exist

#0 /var/www/html/vendor/magento/framework/Code/Reader/ClassReader.php(19): ReflectionClass->__construct('Magento\\Support...')
#1 /var/www/html/vendor/magento/framework/ObjectManager/Definition/Runtime.php(44): Magento\Framework\Code\Reader\ClassReader->getConstructor('Magento\\Support...')
#2 /var/www/html/vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php(71): Magento\Framework\ObjectManager\Definition\Runtime->getParameters('Magento\\Support...')
#3 /var/www/html/vendor/magento/framework/ObjectManager/ObjectManager.php(71): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create('Magento\\Support...')
#4 /var/www/html/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php(163): Magento\Framework\ObjectManager\ObjectManager->get('Magento\\Support...')
#5 /var/www/html/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php(139): Magento\Framework\ObjectManager\Factory\AbstractFactory->parseArray(Array)
#6 /var/www/html/vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php(53): Magento\Framework\ObjectManager\Factory\AbstractFactory->resolveArgument(Array, NULL, Array, 'handlers', 'Magento\\Framewo...')
#7 /var/www/html/vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php(82): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->_resolveArguments('Magento\\Framewo...', Array, Array)
#8 /var/www/html/vendor/magento/framework/ObjectManager/ObjectManager.php(71): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create('Magento\\Framewo...')
#9 /var/www/html/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php(126): Magento\Framework\ObjectManager\ObjectManager->get('Magento\\Framewo...')
#10 /var/www/html/vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php(53): Magento\Framework\ObjectManager\Factory\AbstractFactory->resolveArgument(Array, 'Psr\\Log\\LoggerI...', NULL, 'logger', 'Magento\\Framewo...')
#11 /var/www/html/vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php(82): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->_resolveArguments('Magento\\Framewo...', Array, Array)
#12 /var/www/html/vendor/magento/framework/ObjectManager/ObjectManager.php(71): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create('Magento\\Framewo...')
#13 /var/www/html/vendor/magento/framework/ObjectManager/Factory/AbstractFactory.php(126): Magento\Framework\ObjectManager\ObjectManager->get('Magento\\Framewo...')
#14 /var/www/html/vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php(53): Magento\Framework\ObjectManager\Factory\AbstractFactory->resolveArgument(Array, 'Magento\\Framewo...', NULL, 'logger', 'Magento\\Framewo...')
#15 /var/www/html/vendor/magento/framework/ObjectManager/Factory/Dynamic/Developer.php(82): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->_resolveArguments('Magento\\Framewo...', Array, Array)
#16 /var/www/html/vendor/magento/framework/ObjectManager/ObjectManager.php(57): Magento\Framework\ObjectManager\Factory\Dynamic\Developer->create('Magento\\Framewo...', Array)
#17 /var/www/html/vendor/magento/framework/App/Cache/Frontend/Factory.php(195): Magento\Framework\ObjectManager\ObjectManager->create('Magento\\Framewo...', Array)
#18 /var/www/html/vendor/magento/framework/App/Cache/Frontend/Factory.php(159): Magento\Framework\App\Cache\Frontend\Factory->_applyDecorators(Object(Magento\Framework\Cache\Frontend\Decorator\TagScope))
#19 /var/www/html/vendor/magento/framework/App/Cache/Frontend/Pool.php(67): Magento\Framework\App\Cache\Frontend\Factory->create(Array)
#20 /var/www/html/vendor/magento/framework/App/Cache/Frontend/Pool.php(146): Magento\Framework\App\Cache\Frontend\Pool->_initialize()
#21 /var/www/html/vendor/magento/framework/App/Cache/Type/FrontendPool.php(84): Magento\Framework\App\Cache\Frontend\Pool->get('default')
#22 /var/www/html/vendor/magento/framework/App/Cache/Type/Config.php(49): Magento\Framework\App\Cache\Type\FrontendPool->get('config')
#23 /var/www/html/vendor/magento/framework/Cache/Frontend/Decorator/Bare.php(65): Magento\Framework\App\Cache\Type\Config->_getFrontend()
#24 /var/www/html/vendor/magento/framework/Interception/PluginList/PluginList.php(270): Magento\Framework\Cache\Frontend\Decorator\Bare->load('global|primary|...')
#25 /var/www/html/vendor/magento/framework/Interception/PluginList/PluginList.php(248): Magento\Framework\Interception\PluginList\PluginList->_loadScopedData()
#26 /var/www/html/var/generation/Magento/Framework/App/Http/Interceptor.php(22): Magento\Framework\Interception\PluginList\PluginList->getNext('Magento\\Framewo...', 'launch')
#27 /var/www/html/vendor/magento/framework/App/Bootstrap.php(258): Magento\Framework\App\Http\Interceptor->launch()
#28 /var/www/html/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http\Interceptor))
#29 {main}

So, to sum up, it tries to retrieve not existing class and arguments. Magento Support module doesn't exist same as a class. I can't get where it comes from.

**Comment by TomashKhamlai** [comment]
Hello @ivoinov, I tried to reproduce. Please, try to compare your preferences with [magento/2.1-develop](https://github.com/magento/magento2/blob/2.1-develop/app/etc/di.xml#L181).

Probably, you have Magento\Support\Model\Logger\Handler\Report defined there. Can you check that?
My Redis is fine on 2.1.x and 2.2.x.

**Comment by ivoinov** [comment]
That was the main concern, that I don't have Magento\Support\Model\Logger\Handler\Report defined. 
Not in di.xml, not in the project at all. 
![di_xml](https://user-images.githubusercontent.com/1510448/36304327-eb18a18c-1317-11e8-9169-f8899345ff3e.png)

**Comment by engcom-backlog-nickolas** [comment]
@ivoinov, thank you for your report.
We were not able to reproduce this issue by following the steps you provided. If you'd like to update it, please reopen the issue.

**Comment by magento-engcom-team** [comment]
@ivoinov, thank you for your report.
We were not able to reproduce this issue by following the steps you provided. If you'd like to update it, please reopen the issue.
